web/test/pages/profile.test.tsx

```diff
@@ -0,0 +1,232 @@
+  });
+
+  it("updates the user data on save", async () => {
+    const initialUserData = createEmptyUserData(generateUser({}));
+    const newark = generateMunicipality({ displayName: "Newark" });
+    subject = renderPage({ userData: initialUserData, municipalities: [newark] });
+    fillText("Business name", "Cool Computers");
+    selectByText("Location", newark.displayName);
+    selectByValue("Industry", "e-commerce");
+    selectByValue("Legal structure", "general-partnership");
+    clickSave();
+    await waitFor(() => expect(subject.getByTestId("toast-alert-SUCCESS")).toBeInTheDocument());
+    await waitFor(() => expect(updateSpy).toHaveBeenCalledTimes(1));
+    expect(currentUserData()).toEqual({
+      ...initialUserData,
+      formProgress: "COMPLETED",
+      onboardingData: {
+        ...initialUserData.onboardingData,
+        businessName: "Cool Computers",
+        industry: "e-commerce",
+        homeBasedBusiness: true,
+        legalStructure: "general-partnership",
+        municipality: newark,
+      },
+    });
+  });
+
+  it("builds and sets roadmap on save", async () => {
+    const onboardingData = generateOnboardingData({});
+    const mockSetRoadmap = jest.fn();
+
+    subject = render(
+      withRoadmap(
+        <WithStatefulUserData initialUserData={generateUserData({ onboardingData })}>
+          <Profile displayContent={createEmptyOnboardingDisplayContent()} municipalities={[]} />
+        </WithStatefulUserData>,
+        undefined,
+        mockSetRoadmap
+      )
+    );
+    clickSave();
+    await waitFor(() => expect(updateSpy).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(mockSetRoadmap).toHaveBeenCalledTimes(1));
+  });
+
+  it("prevents user from saving if they have not selected a location", async () => {
+    const newark = generateMunicipality({ displayName: "Newark" });
+    subject = renderPage({ municipalities: [newark] });
+    fillText("Location", "");
+    clickSave();
+    expect(subject.getByTestId("error-alert-REQUIRED_MUNICIPALITY")).toBeInTheDocument();
+    selectByText("Location", newark.displayName);
+    await waitFor(() =>
+      expect(subject.queryByTestId("error-alert-REQUIRED_MUNICIPALITY")).not.toBeInTheDocument()
+    );
+  });
+
+  it("user is able to go back to roadmap", async () => {
+    subject = renderPage({});
+    clickBack();
+    await waitFor(() => expect(mockRouter.mockPush).toHaveBeenCalledWith("/roadmap"));
+  });
+
+  it("prevents user from going back to roadmap if there are unsaved changes", async () => {
+    subject = renderPage({});
+    fillText("Business name", "Cool Computers");
+    clickBack();
+    expect(subject.getByTestId("return")).toBeInTheDocument();
+  });
+
+  it("returns user to profile page from un-saved changes modal", async () => {
+    subject = renderPage({});
+    fillText("Business name", "Cool Computers");
+    clickBack();
+    fireEvent.click(subject.getByTestId("escape"));
```

Comment:
it's small nitpick but where possible, it'd be nicer to use the ProfileDefaults.escapeWhatever as a getByText instead

Reply:
same comment goes for all over this test